NomaiGrandPrix/SpawnActionFactory.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+
+namespace NomaiGrandPrix
+{
+  public class SpawnActionFactory
+  {
+    private static SpawnActionFactory Instance = new SpawnActionFactory();
+
+
+    private SpawnActionFactory()
+    {
+      // Private constructor
+    }
+
+    public static Action[] GetActionsForSpawn(string spawnId)
+    {
+      if (spawnId == null)
+      {
+        throw new InvalidOperationException("Spawn ID for getting actions cannot be null");
+      }
+      
+      switch (spawnId)
+      {
+        case "Spawn_Module_Sunken":
+          return new Action[] { OpenSunkenModuleAirlock };
+        default:
+          return new Action[0];
+      }
```

Comment:
This `switch` seems like it might be better represented as a statically-defined `Dictionary<string, Action[]>` that this code fetch from. Modeling the mapping as data gives us a few benefits:
- It avoids some of the gotchas like forgetting `return` or `break` in the switch statement.
- It opens the possibility of iterating over the dataset, which we may want in the future.

[C#'s collection initializers](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/object-and-collection-initializers?redirectedfrom=MSDN#collection-initializers) would make definition of the dictionary pretty concise. wyt?

Reply:
done